.github/workflows/terraform.yml

```
@@ -0,0 +1,154 @@
+---
+name: Terraform - Linting, formatting, planning, and deployment
+on:
+  # this is for reusable workflows
+  workflow_call:
+    inputs:
+      terraform_workspace:
+        description: "Terraform workspace to deploy to. e.g. dev, stage, prod"
+        required: true
+        type: string
+    secrets:
+      SSH_KEY:
+        description: 'SSH key used to access private repos during the build'
+        required: true
+      GCP_TERRAFORM_SERVICE_ACCOUNT_KEY:
+        description: 'GCP service account credentials json for terraform'
+        required: true
+
```

If SSH_AUTH_SOCK is declared at the workspace level I think it'll be available for all subsequent steps
```suggestion
env:
  SSH_AUTH_SOCK: /tmp/ssh_agent.sock
```